src/wallet/wallet2.cpp

```diff
@@ -4017,7 +4017,7 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
   size_t try_count = 0;
   crypto::hash last_tx_hash_id = m_transfers.size() ? m_transfers.back().m_txid : null_hash;
   std::list<crypto::hash> short_chain_history;
-  tools::threadpool& tpool = tools::threadpool::getInstanceForCompute();
+  tools::threadpool& tpool = tools::threadpool::getInstanceForIO();
   tools::threadpool::waiter waiter(tpool);
```

Comment:
Perhaps these can be renamed to something less generic that points to its usage, like `io_tpool` and `pull_parse_waiter` or something of that nature. The generic naming already present here probably contributed to the original mistake of using the compute threadpool versus the I/O threadpool.

Reply:
Good find!